### PR TITLE
Remove all images for a show, using the ImageClass().

### PR DESCRIPTION
**Pull request by p0psicles**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/server/web/home/handler.py [hunk 1]
```diff
@@ -1793,6 +1793,7 @@ def updateShow(self, show=None):
 
         # force the update
         try:
+            show_obj.remove_images()
```
Comment (p0psicles) [inline medusa/server/web/home/handler.py]: See one line change.

Comment (p0psicles) [reply medusa/server/web/home/handler.py]: Point is, that this is an object oriented approach. As the logic to delete the files in now in the ImageClass, where it should be. To make it more clean, we could even replace some of logic for cleaning up the images after a show delete, using this method.

**Comment by fernandog** [comment]
Ok but as I said it won't work if user only one want to update images and not rescan ALL files.

My other PR has an action in mass update that only update images and doesn't rescan all files from disk

**Comment by p0psicles** [comment]
What's wrong with rescanning? I thought that was the solution, we agreed on? Just do a force full update right?
And if you want to add the massUpdate part. No prob. just use the ImageClass the remove and repopulate the images.

**Comment by fernandog** [comment]
@p0psicles sure but I also want to add an option to only update images. Rescanning a 10 season show with  tons of episodes may be wrong if user want only new images. Understood my pov?

**Comment by p0psicles** [comment]
You can use the Image Class for that. Fine if you add it in this Pr.

**Comment by fernandog** [comment]
@p0psicles ok but I still don't know how to NOT lock the ui when calling a unscheduled task

**Comment by labrys** [comment]
Accidentally pushed to wrong branch.  Reverted after.

**Comment by p0psicles** [comment]
@fernandog just port your changes for the massEdit to this pr.
And the massEdit is a fast action. It's not needed to run in a thread. Threads are used for heavy duty.

Only difference now, compared to your pr. Is that you have to use the command from this PR to remove the images. And after that, you recreate them.

**Comment by fernandog** [comment]
@p0psicles I cherry-picked and changed code BUT its reading ALL files from disk.

I did for only one show with 100 episodes and I got 100 logs like this:
` Setter sets location to`

as you can see in the code:
```python
    @location.setter
    def location(self, value):
        log.debug('{id}: Setter sets location to {location}',
                  {'id': self.series.indexerid, 'location': value})
        self._location = value
        self.file_size = os.path.getsize(value) if value and self.is_location_valid(value) else 0
```

If calls os.path to get size of all files and check if file exists.
It does all that just to update images.
I don't agree with this actions

**Comment by p0psicles** [comment]
I tested this (also the massupdate) and besides one small error, ALl LGTM.

**Comment by fernandog** [comment]
@p0psicles did you read my message about it reading all files from disk when doing this image update?

**Comment by p0psicles** [comment]
Yeah commented on it on slack. But I can't reproduce it. But I didn't have any episodes in my shows. So maybe that's why?

**Pull request merged**